pandas/_typing.py

```diff
@@ -83,8 +83,11 @@
     # Name "npt._ArrayLikeInt_co" is not defined  [name-defined]
     NumpySorter = Optional[npt._ArrayLikeInt_co]  # type: ignore[name-defined]
 
+    from typing_extensions import TypeGuard  # pyright: reportUnusedImport = false
+
```

Comment:
I'd import from standard lib if on python >= 3.10, else from `typing_extensions`.

Reply:
This needs to take into account that `typing_extensions` may not be installed. All in all I'd probably do this something like this:

```python
if PY310:
    from typing import TypeGuard
else:
   try:
        from typing_extensions import TypeGuard
    except ImportError:
        TypeGuard = bool
```

Also add `typing_extensions` to `requirements-dev.txt` , I think it's possible to add it like

`typing_extensions; python_version < '3.10'`

Reply:
does that work in an environment.yml file?  requirements-dev.txt is auto-generated from the environment.yml.

Reply:
checking for py310 requires a circular import which isnt great.  im leaning towards abandoning this PR, would be happy if you'd like to get it across the finish line

Reply:
Oh I didn`t know it is autogenerated and I not knowledgeable with environment.yml to know if it`s possible there. I could take a look. I can also try take over, no problem.